### PR TITLE
Add python 3.9 circle job - let's see how this goes

### PR DESCRIPTION
**Pull request by cicdw**
## Summary

Just wanted to see whether our tests would pass off-the-shelf for Python 3.9!  If so, I'll add a "feature" changelog entry, and if not I'll investigate further.

## Importance

Keeps us up-to-date.

## Checklist

This PR:

- [ ] adds new tests (if appropriate)
- [ ] adds a change file in the `changes/` directory (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Comment by madkinsz** [comment]
I'm looking into this a bit since I want to get multiple python environments setup locally anyway

**Comment by madkinsz** [comment]
> I'm looking into this a bit since I want to get multiple python environments setup locally anyway

I've recreated the error in a local docker image and should be able to put a bit of time into debugging it.

**Comment by cicdw** [comment]
Awesome - let us know what you learn @madkinsz !

**Comment by madkinsz** [comment]
`requirements.txt` installs
`dev-requirements.txt` installs
`test-requirements.txt` installs

`azure` extra fails because no valid `azureml-dataprep-native` version can be found:
```
ERROR: Could not find a version that satisfies the requirement azureml-dataprep-native<14.0.0,>=13.2.0 (from azureml-dataprep[fuse]<1.2.0a,>=1.1.37a->azureml-sdk<1.1,>=1.0.65; extra == "all_extras"->prefect==0.13.9+111.g895e7a591.dirty) (from versions: none)
ERROR: No matching distribution found for azureml-dataprep-native<14.0.0,>=13.2.0 (from azureml-dataprep[fuse]<1.2.0a,>=1.1.37a->azureml-sdk<1.1,>=1.0.65; extra == "all_extras"->prefect==0.13.9+111.g895e7a591.dirty)
```

`spacy` extra fails because it requires `cymem` which does not support `cython>=0.29.x` which seems to be required for `python>=3.9`

`snowflake` extra fails because it requires `pyarrow<0.18.0` which fails building numpy which is required to build pyarrow. Hmm. `numpy` installs just fine otherwise, but is likely using a prebuilt binary.

`pandas` extra fails because `scipy` requires `blas` headers which can be fixed with `sudo apt-get install gfortran libopenblas-dev liblapack-dev`

`dask_cloudprovider` extra fails because no valid `azureml-core` version can be found: `No matching distribution found for azureml-core~=1.16.0 (from azureml-sdk>=1.0.83->dask-cloudprovider<1.0,>=0.2.0 ...)` -- removing the upper limit for `dask_cloudprovider` version does not resolve this.

`great_expectations` extra installs but warns with `great-expectations 0.12.4 requires requests<2.24,>=2.20, but you'll have requests 2.24.0 which is incompatible.`

Running tests fails on collection because of a bad reference in the `feedparser` module.

```
_______________________________________ ERROR collecting tests/tasks/rss/test_feed.py ________________________________________
...
    pytest.importorskip("feedparser")
/usr/local/lib/python3.9/site-packages/feedparser.py:93: in <module>
    _base64decode = getattr(base64, 'decodebytes', base64.decodestring)
E   AttributeError: module 'base64' has no attribute 'decodestring'
```

Bumping to feedparser-6.0.1 fixes this collection error and tests run.

**Comment by madkinsz** [comment]
Test run exceeded CI timeout on `tests/core/test_flow.py::test_timeout_actually_stops_execution[local]` -- this runs fine locally so it'll take a little bit of a longer look.

**Comment by jcrist** [comment]
Currently `dask.distributed` is failing for Python 3.9 - tracking issues here: https://github.com/dask/distributed/issues/4168.

File: tests/conftest.py [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 import tempfile
+import sys
 from unittest.mock import MagicMock
 
 import pytest
-import multiprocessing.popen_spawn_posix  # fix for https://github.com/dask/distributed/issues/4168
+
+if sys.platform != "win32":
+    # Fix for https://github.com/dask/distributed/issues/4168
+    import multiprocessing.popen_spawn_posix  # noqa
```
Comment (jcrist) [inline tests/conftest.py:10]: Why is this needed in `conftest` as well? Shouldn't the patch in the executor be enough?

Comment (madkinsz) [reply tests/conftest.py:10]: Because we import the distributed `Client` on the next line

**Comment by madkinsz** [comment]
Rendered doc addition at https://deploy-preview-3441--prefect-docs.netlify.app/core/getting_started/installation.html#optional-dependencies

**Comment by madkinsz** [comment]
@joshmeek @cicdw this should be ready now

**Comment by cicdw** [comment]
I can't approve since I opened the PR, @joshmeek this one's all you

**Review by joshmeek — APPROVED** [review]
